File: problems/p265.js
var start = +new Date();
var count = 0;

function S(n, r, str) {
    if (str.slice(0, -1).indexOf(str.slice(-n)) !== -1) {
        return 0;
    } else if (r <= 0) {
        var t = str + str.slice(0, n - 1);
        for (var i = 0; i < 2 ** n; i++) {
            if (t.slice(0, i + n - 1).indexOf(t.slice(i, i + n)) !== -1) {
                return 0;
            }
        }
        count += 1;
        return parseInt(str, 2);
    }
    return S(n, r - 1, str + '0') + S(n, r - 1, str + '1')
}

var n = 5;
var sol = S(n, 2 ** n - n, '0'.repeat(n));
console.log('Solution: ' +  sol)
console.log('Count: ' + count);

var end = +new Date();
console.log((end - start)/1000 + ' seconds to run');

File: problems/201-300/p265.java
import java.util.Arrays;
import java.util.Collections;

public class p265 {

    static int count = 0;
    static int n = 5;
    static int stringSize = (int) Math.pow(2, n);

    static long S(int n, int r, String str) {
        String head = str.substring(0, str.length() - 1);
        String tail = str.substring(str.length() - n);
        if (head.contains(tail)) {
            return 0;
        } else if (r <= 0 ) {
            String t = str + str.substring(0, n);
            for (int i = 0; i < stringSize; i++) {
                head = t.substring(0, i + n - 1);
                tail = t.substring(i, i + n);
                if (head.contains(tail)) {
                    return 0;
                }
            }
            count += 1;
            return Integer.parseInt(str, 2);
        }
        return S(n, r - 1, str + "0") + S(n, r - 1, str + "1");
    }

    public static void main(String[] args) {
        long start = System.nanoTime();

        String baseString = String.join("", Collections.nCopies(n, "0"));
        long sol = S(n, stringSize - n, baseString);

        System.out.println("Solution: " + sol);
        System.out.println("Count: " + count);

        long end = System.nanoTime();
        System.out.println((end - start)/1000000000.0 + " seconds to run");
    }
}
